pkg/unittest/assertion.go

```diff
@@ -70,7 +70,8 @@ func (a *Assertion) Assert(
 		singleTemplateResult[template] = rendered
 
 		// Update the DocumentIndex if the found idx is not -1
-		indexError := a.determineDocumentIndex(singleTemplateResult)
+		selectedDocs, indexError := a.selectDocuments(rendered)
```

Comment:
Ah, actually - we should check if selection generates more than 1 result across all template files. I guess it needs improving.

Reply:
Addressed it.